core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -219,24 +250,34 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
                 if not newest_version or Version(newest_version) < Version(semver_version):
                     newest_version = semver_version
             if not newest_version:
+                logger.error(f"No firmware versions found for this configuration ({vehicle} and {platform}).")
                 return None
             version = f"STABLE-{newest_version}"
 
+        firmware_format = FirmwareDownload._supported_firmware_formats[platform]
+
         items = self._find_version_item(
             vehicletype=vehicle.value,
             platform=platform.value,
             mav_firmware_version_type=version,
-            format=FirmwareDownload._supported_firmware_format,
+            format=firmware_format,
         )
 
         if len(items) != 1:
-            logger.error(f"Invalid number of candidates to download: {len(items)}")
+            logger.error(f"Invalid number of candidates to download ({len(items)}): {items}")
             return None
 
         item = items[0]
+        logger.debug(f"Downloading following firmware: {item}")
         path = FirmwareDownload._download(item["url"])
-        if not path or not FirmwareDownload._validate_firmware(path):
-            logger.error("Unable to validate firmware file.")
-            return None
+        if firmware_format == FirmwareFormat.ELF:
+            # Make the binary executable
+            os.chmod(str(path), stat.S_IXOTH)
+
+        # TODO: Reincorporate firmware validation for SITL once Ardupilot fixes non-zero return on --help (PR #17573)
+        if platform != Platform.SITL:
```

Comment:
Why are you not validating SITL ?

Reply:
?

Reply:
My bad, I just accepted and added my comment there.